src/test/java/org/folio/entitlement/it/KeycloakRetriesIT.java

```diff
@@ -0,0 +1,212 @@
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Primary;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlMergeMode;
+
+@IntegrationTest
+@SqlMergeMode(MERGE)
+@Sql(executionPhase = AFTER_TEST_METHOD, scripts = "classpath:/sql/truncate-tables.sql")
+@TestPropertySource(properties = {
+  "application.keycloak.enabled=true",
+  "application.okapi.enabled=false",
+  "application.kong.enabled=false",
+  "application.clients.folio.connect-timeout=250ms",
+  "application.clients.folio.read-timeout=250ms",
+  "retries.module.backoff.delay=1",
+  "retries.module.backoff.multiplier=1",
+  "application.keycloak.admin.clientId=mgr-component-app"
+})
+@Import(KeycloakRetriesIT.TestConfiguration.class)
+class KeycloakRetriesIT extends BaseIntegrationTest {
+
+  private static final String FOLIO_APP1_ID = "folio-app1-1.0.0";
+  private static final String FOLIO_APP2_ID = "folio-app2-2.0.0";
+
+  @BeforeEach
+  public void initKeycloakMock() throws Exception {
+    var tokenResponse = new AccessTokenResponse();
+    tokenResponse.setToken("mock_token");
+    tokenResponse.setExpiresIn(6000000);
+    stubPost(getWireMockClient(), 1, urlEqualTo("/realms/master/protocol/openid-connect/token"), tokenResponse, 200);
+  }
+
+  @Test
+  @WireMockStub(scripts = {"/wiremock/mgr-tenants/test/get.json",
+    "/wiremock/mgr-applications/folio-app1/get-by-ids-query-full.json",
+    "/wiremock/mgr-applications/folio-app1/get-discovery.json",
+    "/wiremock/mgr-applications/validate-any-descriptor.json", "/wiremock/folio-module1/install.json"})
+  void verifyModuleRegistrationRetry() throws Exception {
+    var moduleId = "folio-module1-1.0.0";
+    var wireMockClient = mockBaseDependencies(moduleId, "route123");
+
+    var entitlementRequest = entitlementRequest(FOLIO_APP1_ID);
+    var queryParams = Map.of("tenantParameters", "loadReference=true", "ignoreErrors", "true");
+    var request =
+      post(updatePathWithPrefix("/entitlements")).contentType(APPLICATION_JSON).header(TOKEN, getSystemAccessToken())
+        .content(asJsonString(entitlementRequest));
+    queryParams.forEach(request::queryParam);
+
+    stubPost(wireMockClient, 1, urlEqualTo("/admin/realms/test/clients/null/authz/resource-server/scope"), null, 201);
+    stubAnyHttpMethod(wireMockClient, 1,
+      urlMatching("/admin/realms/test/clients/null/authz/resource-server/resource.*"), null, 500);
+
+    mockMvc.perform(request).andExpect(status().isBadRequest()).andExpect(
+      jsonPath("$.errors[0].parameters[0].value").value(
+        containsString("Failed to update authorization resources in Keycloak")));
+
+    var endpointsCalled =
+      wireMockClient.getServeEvents().stream().filter(e -> e.getResponse().getStatus() == 500).toList();
+    assertEquals(3, endpointsCalled.stream().filter(
```

Comment:
we are using assertJ library for assertions: https://joel-costigliola.github.io/assertj/